### PR TITLE
[ENH] Add native implementation of PatchTST for long-term forecasting

### PR DESCRIPTION
**Pull request by nareshmethuku**
This PR introduces the **PatchTST (A Time Series is Worth 64 Words)** architecture to `pytorch-forecasting`. PatchTST leverages Channel Independence and patching of time series to achieve state-of-the-art performance on long-term forecasting.

### Changes 
- Implements `PatchTST` and core submodules (`PatchEmbedding`, `FlattenHead`).
- Inherits from `BaseModelWithCovariates` with full support for static, time-varying continuous, and time-varying categorical covariates. Following the principles of Channel Independence, all covariates are mapped into independent channels.
- Full support for probabilistic loss functions like `QuantileLoss` via multi-quantile forecasting in the flattening head.
- Includes comprehensive integration and unit tests for the architecture checking diverse dataloader and multi-target setups.
- Updated `README.md` to list PatchTST as a supported model.

### Tests
Tested via `pytest tests/test_models/test_patch_tst.py` under various configurations:
- `test_integration` (standard MAE training loop)
- `test_quantile_loss`
- `test_multiple_targets`
- `test_no_exogenous_variables`

All tests pass locally with full coverage. Code is formatted via `pre-commit`.

### CODE REVIEW

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks a lot for the PR and welcome to `pytorch-forecasting`!
Can you please also implement `pkg` class for `PatchTST`? (See [`DeepAR_pkg`](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/models/deepar/_deepar_pkg.py), or any other model `pkg` class for example) This will be used to test the model over a fixed set of test scenarios by the unified test framework.

See the `pkg` class doc here: https://pytorch-forecasting.readthedocs.io/en/stable/pkg.html

Also can you please move the `sub_modules` to `layers` module?

**Comment by phoeenniixx** [comment]
A suggestion:
We are working on a v2 rework of the library (see the docs for v2 API [here](https://pytorch-forecasting.readthedocs.io/en/stable/api_v2.html))
v2 is expected to make the package more user friendly and reduce the coupling in classes
Would you be interested in implementing this model in v2 as well? 

Also, it would be greatly appreciated if you could give us some feedback on v2 - try out the tutorials [here](https://pytorch-forecasting.readthedocs.io/en/stable/tutorials_v2.html).
You can also suggest some feats that you would like to have in v2 (as we are still in development phase of the API, it is quite flexible :) )

**Comment by nareshmethuku** [comment]
Hi @phoeenniixx,  I've addressed all the requested changes, updated the branch, and all checks are passing now. Could you please take another look when you have a chance? Also looping in @fkiraly  @benHeid  @jdb78  for review if available. Thank you!

File: README.md [hunk 1]
```diff
@@ -99,39 +106,50 @@ training_cutoff = "YYYY-MM-DD"  # day for cutoff
...
+    static_reals=[...],
     # covariates known and unknown in the future to inform prediction
-    time_varying_known_categoricals=[ ... ],
-    time_varying_known_reals=[ ... ],
-    time_varying_unknown_categoricals=[ ... ],
-    time_varying_unknown_reals=[ ... ],
+    time_varying_known_categoricals=[...],
+    time_varying_known_reals=[...],
+    time_varying_unknown_categoricals=[...],
+    time_varying_unknown_reals=[...],
```
Comment (phoeenniixx) [inline README.md]: Why are we changing the example code?

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks!
There are a lot of unrelated changes in the PR can you please revert them?

Also, for v2 I meant a new PR, but np we can do both of these in one go as well

File: pytorch_forecasting/models/patch_tst/_patch_tst_pkg_v2.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+"""PatchTST package container for V2."""
+
+from pytorch_forecasting.models.base._base_object import _BasePtForecaster
+
+
+class PatchTSTV2_pkg(_BasePtForecaster):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/patch_tst/_patch_tst_pkg_v2.py]: please use `BasePkg` for v2

File: pytorch_forecasting/models/patch_tst/_patch_tst_pkg_v2.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                "data_loader_kwargs": dict(
+                    target_normalizer=GroupNormalizer(
+                        groups=["agency", "sku"], transformation="softplus"
+                    ),
+                ),
+            },
+        ]
+
+    @classmethod
+    def _get_test_dataloaders_from(cls, params):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/patch_tst/_patch_tst_pkg_v2.py]: we dont need this method for v2!

File: pytorch_forecasting/models/patch_tst/_patch_tst_v2.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+import warnings as warn
+
+import torch
+import torch.nn as nn
+from torch.optim import Optimizer
+
+from pytorch_forecasting.models.base._tslib_base_model_v2 import TslibBaseModel
+
+
+class PatchTSTV2(TslibBaseModel):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/patch_tst/_patch_tst_v2.py]: Is this a model interfaced from [`tslib`](https://github.com/thuml/Time-Series-Library)?
If not, you can also try  [`BaseModel`](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/models/base/_base_model_v2.py#L24) and the compatible datamodule as [`EncoderDecoderTimeSeriesDataModule`](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/data/data_module/_encoder_decoder_data_module.py#L35)

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks!
- Please use [`BasePkg`](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/base/_base_pkg.py#L16) for v2 
- also add the compatible data module to the pkg using `get_datamodule_cls`

Few more comments see below

**Review by phoeenniixx — APPROVED** [review]
Thanks a lot for the contribution and help @nareshmethuku !
Just updated the names and added API refernce for v1 model as well
(Will merge once all the tests pass!)

If you want to chat or suggest something that you would like to have in v2, please feel free to raise an issue or just ping us on discord: https://discord.gg/7eTMmuWug
